File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int n=ransomNote.size();     // Len of the Ransom Note String
        int m=magazine.size();       // Len of the Magazine String

        unordered_map<char,int> need;   // Declaration of the Need Hashmap (Ransom Note)
        unordered_map<char,int> have;   // Declaration of the Have Hashmap (Magazine)

        for(int i=0;i<n;i++){
            need[ransomNote[i]]++;
        }

        for(int i=0;i<m;i++){
            have[magazine[i]]++;
        }

        return fun(have,need);
    }

    bool fun(unordered_map<char,int>have,unordered_map<char,int>need){
        for(auto i:need){
            char c=i.first;      // Key one 
            int fneed=i.second;  // Value One 
            int fhave=have[c];   //The freq of that letter in have

            if(fhave<fneed){
                return false;
            }
        }
        return true;
    }
};

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        n=len(ransomNote)        # Len of the Ransom Note String
        m=len(magazine)          # Len of the Magazine String

        need={}                  # Declaration of the Need HashMap (Ransom Note)
        have={}                  # Declaration of the Have HashMap (Magazine)

        for i in range(n):
            need[ransomNote[i]]=need.get(ransomNote[i], 0) + 1

        for i in range(m):
            have[magazine[i]]=have.get(magazine[i], 0) + 1

        return self.fun(have,need)

    def fun(self,have,need):
        for c, fneed in need.items():
            # c -> Key One
            # fneed -> Value One
            fhave = have.get(c, 0)    # The freq of that letter in have

            if fhave<fneed:
                return False

        return True